ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -244,15 +246,20 @@ class WebSocketService(
 
   private val numConns = new java.util.concurrent.atomic.AtomicInteger(0)
 
+  @volatile private var lastSeenOffset: Option[JsValue] = None
```

Comment:
The `WebSocketService` is not an appropriate scope for this.

Reply:
why? it is one instance per client connection, so should be all right.

Reply:
it is defined in the class, not in the companion object.

Reply:
No it is not. Note the `numConns` variable immediately adjacent, or

https://github.com/digital-asset/daml/blob/3a79143b98202d656822cd44fdb9d04b5f83bc03/ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala#L143

This confusion is only further evidence that "within the stream" is the minimum scope safety required for this feature.

Reply:
yes, you are right... switching it to `Flow.scan`